### PR TITLE
[ADD] provelo_custom_invoice_auto_open

### PR DESCRIPTION
**Pull request by robinkeunen**
## Description

Invoices are automatically opened for users with "Automatically Open Invoices" group set.

## Odoo task (if applicable)

[TASK](https://gestion.coopiteasy.be/web#id=7962&action=479&model=project.task&view_type=form&menu_id=536)

## Checklist before approval

- [x] Tests are present (or not needed).
- [x] Credits/copyright have been changed correctly.
- [x] Change log snippet is present.
- [x] (If a new module) Moving this to OCA has been considered.

=> The OCA does not like automatic invoice validation.

### CODE REVIEW

File: provelo_custom_invoice_auto_open/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.multi
+    def action_invoice_create(self, grouped=False, final=False):
+        invoice_ids = super().action_invoice_create(grouped=grouped, final=final)
+        account_invoice_auto_open_group_id = self.env.ref(
+            "provelo_custom_invoice_auto_open.account_invoice_auto_open_group"
+        )
+        if account_invoice_auto_open_group_id.id in self.env.user.groups_id.ids:
```
Comment (huguesdk) [inline provelo_custom_invoice_auto_open/models/sale_order.py]: why not:
```suggestion
        if self.env.user.has_group(
            "provelo_custom_invoice_auto_open.account_invoice_auto_open_group"
        ):
```
as you suggested in the task?

File: provelo_custom_invoice_auto_open/models/res_config_settings.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    group_account_invoice_auto_open = fields.Boolean(
+        string="Automatically Open Invoices",
+        implied_group="provelo_custom_invoice_auto_open"
+        ".account_invoice_auto_open_group",
+    )
```
Comment (huguesdk) [inline provelo_custom_invoice_auto_open/models/res_config_settings.py]: as i understand, either this setting is true and the invoices open automatically for all employees, or it is false, and invoices open automatically only for members of the `provelo_custom_invoice_auto_open.account_invoice_auto_open_group` group. is this correct?

i think this should be explained in the readme. (and maybe even add a test for this?)

**Review by huguesdk — APPROVED** [review]
`res.config.settings`, tests and all. impressive! 👍 lgtm apart from the 2 remarks.

File: provelo_custom_invoice_auto_open/models/sale_order.py (hunk 1 above)
Comment (robinkeunen) [reply provelo_custom_invoice_auto_open/models/sale_order.py]: indeed why ? fixed

File: provelo_custom_invoice_auto_open/models/res_config_settings.py (hunk 2 above)
Comment (robinkeunen) [reply provelo_custom_invoice_auto_open/models/res_config_settings.py]: It is tested in 

- `test_create_invoice_opens_it_for_auto_invoice_group`
- `test_create_invoice_does_not_open_it_without_auto_invoice_group`

But I can add a comment.

**Review by carmenbianca — APPROVED** [review]
LGTM

**Pull request merged**